matrix/build.gradle

```diff
@@ -18,53 +18,85 @@ dependencies {
   implementation("io.opentelemetry:opentelemetry-extension-annotations:${versions["opentelemetry"]}")
 }
 
-def dockerWorkingDir = new File(project.buildDir, "docker")
-
 def buildProprietaryTestImagesTask = tasks.create("buildProprietaryTestImages") {
   group = "build"
   description = "Builds all Docker images for the test matrix for proprietary app servers"
 }
 
-def targets = [
-   "weblogic" : [
+def proprietaryTargets = [
+    "weblogic" : [
        "12.1.3" : ["developer"],
        "12.2.1.4" : ["developer"],
         "14.1.1.0" : ["developer-8", "developer-11"]
-   ],
+    ],
     "jboss-eap" : [
         "7.1.0" : ["8"],
         "7.3.0" : ["8", "11"]
     ]
 ]
 
-targets.forEach { server, data ->
-  String dockerFileName = "${server}.dockerfile"
-  data.forEach { version, jdks ->
-    jdks.forEach { jdk ->
-      def prepareTask = tasks.register("${server}ImagePrepare-$version-jdk$jdk", Copy) {
-        def warTask = project.tasks.named("war").get()
-        it.dependsOn(warTask)
-        it.into(dockerWorkingDir)
-        it.from("src/${dockerFileName}")
-        it.from("src/main/docker/${server}")
-        it.from(warTask.archiveFile) {
-          rename { _ -> "app.war" }
+createDockerTasks(buildProprietaryTestImagesTask, proprietaryTargets, false)
+
+def buildWindowsTestImagesTask = tasks.create("buildWindowsTestImages") {
+  group = "build"
+  description = "Builds all Windows Docker images for the test matrix"
+}
+
+def windowsTargets = [
+    "tomcat" : [
+        [version: "7.0.107", majorVersion: "7"]: ["8"],
+        [version: "8.5.61", majorVersion: "8"]: ["8", "11"],
+        [version: "9.0.41", majorVersion: "9"]: ["8", "11"]
+    ],
+    "jetty" : [
+        [version: "9.4.35", sourceVersion: "9.4.35.v20201120"]: ["8", "11", "15"],
+        [version: "10.0.0", sourceVersion: "10.0.0.beta3", classifier: "split"]: ["11", "15"]
```

Comment:
What is this `split` and why do we need this?

Reply:
`classifier` is used to be able to choose a different `Dockerfile` without changing the name of the appserver. Different versions may require setup that is different enough that they require a separate `Dockerfile`, yet it makes no sense to have the final image have a different name just because of that.

Reply:
In this specific case, it causes the `jetty-split.windows.dockerfile` to be used instead, which is used for Jetty versions where HOME and BASE must be split to two different locations, and the download URLs for those are also different, so separate Dockerfile made sense.

Reply:
Why do we need to test these two Jetty setups? Do we expect them to behave differently with javaagent?

Reply:
No. They are not different setups, the required directory layout and download URLs were just changed starting from version 10. I did not want to expose this internal detail to the final Docker image, therefore the classifier argument, which does not leak to the final image name, is used instead to specify that a different Dockerfile should be used instead. Otherwise there would have to be a separate "jetty-10" server entry, which would cause the image name to become "jetty-10-10.0.0" which would not be nice.